nipype/interfaces/utility/nifti.py

```diff
@@ -0,0 +1,186 @@
+import numpy as np
+import nibabel as nb
+
+from io import open
+
+from nipype import logging
+from nipype.utils import NUMPY_MMAP
+from nipype.interfaces.base import traits, TraitedSpec, BaseInterface, BaseInterfaceInputSpec, File
+
+
+IFLOGGER = logging.getLogger('interface')
+
+DTYPE_MAP = {
+    1: np.uint8, 2: np.uint8, 256: np.uint8,
+    4: np.uint16,
+    8: np.uint32, 1024: np.uint32, 1280: np.uint32,
+    64: np.float32, 1536: np.float32
+}
+
+class ConformImageInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc='input image')
+    check_ras = traits.Bool(True, usedefault=True,
+                            desc='check that orientation is RAS')
+    check_dtype = traits.Bool(False, usedefault=True,
+                              desc='check data type, and limit to 8-32 bits')
+    base_affine = traits.Bool(False, usedefault=True,
+                              desc='ignore sform and qform matrices')
+    save_hdr = traits.Bool(False, usedefault=True, desc='backup old header')
+
+
+class ConformImageOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc='output conformed file')
+    out_hdr = File(desc='Python pickle containing backed-up header')
+
+
+class ConformImage(BaseInterface):
+
+    """
+    Conforms the orientation, affine matrix and data type of the input image
+
+    """
+    input_spec = ConformImageInputSpec
+    output_spec = ConformImageOutputSpec
+
+    def __init__(self, **inputs):
+        self._results = {}
+        super(ConformImage, self).__init__(**inputs)
+
+    def _list_outputs(self):
+        return self._results
+
+    def _run_interface(self, runtime):
+        # load image
+        if not np.any((self.inputs.check_ras,
+                       self.inputs.check_dtype,
+                       self.inputs.base_affine)):
+            self._results['out_file'] = self.inputs.in_file
+            IFLOGGER.warning('No conform operation requested on the input image "%s".',
+                             self.inputs.in_file)
+            return runtime
+
+        # Generate name
+        out_file, ext = op.splitext(op.basename(self.inputs.in_file))
+        if ext == '.gz':
+            out_file, ext2 = op.splitext(out_file)
+            ext = ext2 + ext
+
+        nii = nb.load(self.inputs.in_file, mmap=NUMPY_MMAP)
+        data = nii.get_data()
+        hdr = nii.header.copy()
+
+        if self.inputs.save_hdr:
+            self._results['out_hdr'] = op.abspath('{}_hdr.pklz'.format(out_file))
+            with open(self._results['out_hdr'], 'wb') as fheader:
+                hdr.write_to(fheader)
+
+
+        aff = nii.affine
+
+        if self.inputs.check_ras:
+            data = nb.as_closest_canonical(nii).get_data()
+
+        if self.inputs.base_affine:
+            data = nb.as_closest_canonical(nii).get_data()
+            aff = nb.as_closest_canonical(nii).header.get_base_affine()
+            hdr['qform_code'] = 0
+            hdr['sform_code'] = 0
```

Comment:
my worry is with this setting. this brings us back to analyze days. if i'm given this file without any context, then i have no way of knowing what's left or right in the brain.

wouldn't it be better to use an sform code that aligns with a standard?

https://nifti.nimh.nih.gov/nifti-1/documentation/faq#Q18